File: llm_generated_code/ChatGPT/CPP/Level1/Q30.cpp
#include <iostream>
#include <string>
#include <algorithm>

using namespace std;

int minOperationsToAllOnes(const string& s) {
    int count0 = count(s.begin(), s.end(), '0');
    if (count0 == 0) return 0;
    if (count0 == 4) return -1;
    if (count0 == 3) return 3;
    if (count0 == 2) {
        if ((s[0] == '0' && s[1] == '0') || (s[1] == '0' && s[2] == '0') || (s[2] == '0' && s[3] == '0'))
            return 2;
        return 3;
    }
    return 1;
}

int main() {
    int t;
    cin >> t;
    while (t--) {
        string s;
        cin >> s;
        cout << minOperationsToAllOnes(s) << endl;
    }
    return 0;
}

File: llm_generated_code/ChatGPT/Python/Level1/Q30.py
def min_operations_to_all_ones(s):
    if s.count("1") == 4:
        return 0
    elif s.count("0") == 4:
        return -1
    elif s.count("1") == 3:
        return 1
    elif s.count("1") == 2:
        return 2
    return 3

t = int(input())
results = []
for _ in range(t):
    s = input()
    results.append(str(min_operations_to_all_ones(s)))
print("\n".join(results))
